core/src/pending_collision.rs

```diff
@@ -0,0 +1,240 @@
+            let vertices = match mesh.attribute(Mesh::ATTRIBUTE_POSITION).unwrap() {
+                VertexAttributeValues::Float32x3(vertices) => vertices,
+                _ => unreachable!(
+                    "Mesh should have encoded vertices as VertexAttributeValues::Float32x3"
+                ),
+            };
+            let mut points = Vec::with_capacity(vertices.len());
+            for vertex in vertices {
+                points.push(Vec3::new(vertex[0], vertex[1], vertex[2]));
+            }
+            commands
+                .entity(*child)
+                .insert(pending_collision.body_type)
+                .insert(CollisionShape::ConvexHull {
+                    points,
+                    border_radius: pending_collision.border_radius,
+                });
+        }
+    }
+
+    generated
+}
+
+#[cfg(test)]
+mod tests {
+    use bevy::{
+        asset::AssetPlugin,
+        core::CorePlugin,
+        gltf::GltfPlugin,
+        pbr::PbrPlugin,
+        prelude::*,
+        render::{options::WgpuOptions, RenderPlugin},
+        scene::ScenePlugin,
+        window::WindowPlugin,
+    };
+
+    use super::*;
+
+    // Allows run tests for systems containing rendering related things without GPU
+    pub(super) struct HeadlessRenderPlugin;
+
+    impl Plugin for HeadlessRenderPlugin {
+        fn build(&self, app: &mut App) {
+            app.insert_resource(WgpuOptions {
+                backends: None,
+                ..Default::default()
+            })
+            .add_plugin(CorePlugin::default())
+            .add_plugin(TransformPlugin::default())
+            .add_plugin(WindowPlugin::default())
+            .add_plugin(AssetPlugin::default())
+            .add_plugin(ScenePlugin::default())
+            .add_plugin(RenderPlugin::default())
+            .add_plugin(PbrPlugin::default())
+            .add_plugin(GltfPlugin::default());
+        }
+    }
+
+    #[test]
+    fn pending_collision_assignes() {
+        let mut app = App::new();
+        app.add_plugin(HeadlessRenderPlugin)
+            .add_system(pending_collision_system);
+
+        const REQUESTED_COLLISION: PendingConvexCollision = PendingConvexCollision {
+            body_type: RigidBody::Static,
+            border_radius: None,
+        };
+
+        let parent = app
+            .world
+            .spawn()
+            .insert(Transform::default())
+            .insert(GlobalTransform::default())
+            .insert(REQUESTED_COLLISION)
+            .id();
+
+        app.world
+            .resource_scope(|world, mut scene_spawner: Mut<'_, SceneSpawner>| {
+                let asset_server = world.get_resource::<AssetServer>().unwrap();
+                scene_spawner.spawn_as_child(asset_server.load("cubes.glb#Scene0"), parent);
+                scene_spawner
+                    .spawn_queued_scenes(world)
+                    .expect("Unable to spawn scene");
+            });
```

Comment:
Would it be possible to not load a real asset? Could we generate a very simple mesh from code and add it to the assets?

I think that would: 
* remove the need to first open the asset in blender to know what is the expected output.
* remove the need to commit the assets
* remove the need for many bevy plugins to be used for the tests (the fewer dependencies, the better)

after all, this feature is related to meshes, not to gltf. It should work fine regardless of how the mesh is imported/created right?

Reply:
Right... I will think how I can improve it.

Reply:
If you need help, I may also look at it. But not today, I don't have time. Maybe tomorrow.

Reply:
Done! You are right, it much better.